tests/unit/test_adapter_registry.py

```diff
@@ -1,11 +1,28 @@
-from openscm.adapters import _loaded_adapters, load_adapter
+import re
+
+
 import pytest
+from unittest.mock import patch
+
+
+from openscm.adapters import load_adapter
+from openscm.errors import AdapterNeedsModuleError
 
 
+@patch("openscm.adapters._loaded_adapters", new={"stub": 1})
 def test_adapter_registry():
-    assert _loaded_adapters == {}
-    stub = 1
-    _loaded_adapters["stub"] = stub
-    assert load_adapter("stub") == stub
+    assert load_adapter("stub") == 1
+
+
+def test_adapter_registry_unknown_model():
     with pytest.raises(KeyError, match="Unknown model 'unknown'"):
         load_adapter("unknown")
+
+
+def test_adapter_registry_import_error():
+    error_msg = re.escape(
```

Comment:
Why `re.escape`?

Reply:
from memory, pytests's match needs to have things escaped if you special characters like square brackets and spaces and new lines